File: deepspace/src/deepspace/ShieldBooster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author antonio
 */
public class ShieldBooster {
    
    private String name;
    private float boost;
    private int uses;
    
    ShieldBooster(String name, float boost, int uses){
        this.name=name;
        this.boost=boost;
        this.uses=uses;
    }
    
    ShieldBooster(ShieldBooster s){
        this(s.name, s.getBoost(), s.getUses());
    }
    
    public float getBoost(){
        return boost;
    }

    public int getUses(){
        return uses;
    }
    
    public float useIt(){
        if( uses > 0 ){
            uses--;
            return boost;
        }
        else
            return 1.0f;
    }
}

File: DeepSpace/lib/ShieldBooster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'ShieldToUI'

module Deepspace
  class ShieldBooster
    def initialize(name, boost, uses)
      @name=name
      @boost=boost
      @uses=uses
    end
    
    def self.newCopy(s)
     new(s.name, s.boost, s.uses)
    end

    attr_reader :boost, :uses
    
    def useIt
      if @uses > 0
        @uses-=1
        @boost
      else
        1.0
      end
    end
    
    def getUIversion
      ShieldToUI.new(self)
    end
    
    def to_s
      "Name: #{@name} Boost: #{@boost} Uses: #{@uses}"
    end
    
  end
end
